### PR TITLE
Index and show for categories

### PR DESCRIPTION
**Pull request by neerajdotname**
show for category https://monosnap.com/file/74vYM1ftVdklG6z2wnjZIQX4h6FjGk

index for all categories https://monosnap.com/file/gdAEO21oEOdrZrTKwoFmNgcJX9U53J

This PR should be looked at after PR https://github.com/bigbinary/acehelp/pull/176 is merged.

### CODE REVIEW

**Comment by vipulnsward** [comment]
@breezyboa _a

**Comment by breezyboa** [comment]
@sudeeptarlekar _a

File: app/graphql/mutations/article_mutations.rb [hunk 1]
```diff
@@ -94,4 +94,46 @@ class Mutations::ArticleMutations
       }
     }
   end
+
+  Upvote = GraphQL::Relay::Mutation.define do
+    name "Upvote"
+
+    input_field :id, !types.ID
```
Comment (sudeeptarlekar) [inline app/graphql/mutations/article_mutations.rb]: Please change type to string as we are using uuids now

File: app/graphql/mutations/article_mutations.rb [hunk 2]
```diff
@@ -94,4 +94,46 @@ class Mutations::ArticleMutations
...
+      else
+        GraphQL::ExecutionError.new("Invalid input: #{article.errors.full_messages.join(', ')}")
+      end
+    }
+  end
+
+  Downvote = GraphQL::Relay::Mutation.define do
+    name "Downvote"
+
+    input_field :id, !types.ID
```
Comment (sudeeptarlekar) [inline app/graphql/mutations/article_mutations.rb]: change id to string

File: app/javascript/Client/Data/Article.elm [hunk 3]
```diff
@@ -31,9 +32,48 @@ type alias ArticleSummary =
 
 
 -- ENCODERS
+
+
+encodeUpvote : ArticleId -> Encode.Value
+encodeUpvote articleId =
+    let
+        query =
+            "mutation upvote {upvoteArticle(input: {id: " ++ (toString articleId) ++ "}) {article {id upvotes_count}}}"
```
Comment (sudeeptarlekar) [inline app/javascript/Client/Data/Article.elm]: once you converted ids to string no need of parsing again

File: app/javascript/Client/Data/Article.elm [hunk 4]
```diff
@@ -31,9 +32,48 @@ type alias ArticleSummary =
...
+            , ( "query", Encode.string query )
+            , ( "variables", Encode.object [] )
+            ]
+
+
+encodeDownvote : ArticleId -> Encode.Value
+encodeDownvote articleId =
+    let
+        query =
+            "mutation downvote { downvoteArticle(input: {id: " ++ (toString articleId) ++ "}) {article {id downvotes_count } }}"
```
Comment (sudeeptarlekar) [inline app/javascript/Client/Data/Article.elm]: same

File: db/migrate/20180704122137_add_downvotes_to_articles.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddDownvotesToArticles < ActiveRecord::Migration[5.2]
+  def change
+    add_column :articles, :downvotes_count, :int, default: 0
+  end
+end
```
Comment (sudeeptarlekar) [inline db/migrate/20180704122137_add_downvotes_to_articles.rb]: Please add a missing new line 

Thanks in advance

**Comment by neerajdotname** [comment]
Can we stop such feedback about missing new lines etc.

My distaste is because it should be taken care of by formatted. Until Ruby does not get formatter I’m ok with a bit of inconsistency.

So going forward whatever rubocop autofixes is good. What is not fixed I’m ok with it until the day Ruby gets a formatter.

**Comment by sudeeptarlekar** [comment]
@sudeeptarlekar _a

**Comment by sudeeptarlekar** [comment]
@vipulnsward _a please review

**Pull request merged**